rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,503 @@
+
+This involves requesting a token by making a `POST` request to `https://webhook.site/token` with the following parameters.
+```json
+{
+  "default_status": 200,
+  "default_content": "Hello world!",
+  "default_content_type": "text/html",
+  "timeout": 0
+}
+```
+The response contains a **uuid**, which can be used in conjunction with the base url, to access the webhook endpoint at https://webhook.site/{token.uuid}
+```json
+{
+  "redirect": false,
+  "alias": null,
+  "timeout": 0,
+  "premium": true,
+  "uuid": "9981f9f4-657a-4ebf-be7c-1915bedd4775",
+  "ip": "127.0.0.1",
+  "user_agent": "Paw\/3.1.8 (Macintosh; OS X\/10.14.6) GCDHTTPRequest",
+  "default_content": "Hello world!",
+  "default_status": 200,
+  "default_content_type": "text\/plain",
+  "premium_expires_at": "2019-10-22 10:52:20",
+  "created_at": "2019-09-22 10:52:20",
+  "updated_at": "2019-09-22 10:52:20"
+}
+```
+
+An Exmaple of alerts that were triggered from Prometheus (as part of my POC) can be found [here](https://webhook.site/#!/c21467b9-7ce4-4389-998b-8d91578433f3/ff66d8b1-c206-49a7-b5f8-b44af28a022f/1)
+
+> Privacy can be a concern here as some users may not want their private alerts to go to a public webhook. The default behaviour can be easily overridden using the `config.yaml` file passed to the `--addon-config` flag as shown below
+
+```yaml
+publicWebhook: false 
+...
+Additional Configuration Items.
+...
+```
+
+---
+- ## Installation YAML
+
+The user can pass a yaml file along with the `--addon-config` flag to override the default settings we provide.
+```sh
+linkerd install --addon-config config.yaml | kubectl apply -f -
+```
+
+The yaml file will be unmarshalled during installation and will overwrite the default values in `values.yaml`
+
+The template of the `yaml` file can be defined as follows 
+
+```yaml
+# Flag to enable/disable sending alerts to Webhook.site
+publicWebhook: true [ default = false ]
+
+# This section contains configuration items specific to Alertmanager addon component
+alertmanager:
+
+  # Name of the Alertmanager component, re-used across helm chats 
+  name: alertmanager [ default = alertmanager ]
+
+  # Flag to determine if Alertmanager should be enabled/disabled.
+  enabled: true [ default = false ]
+
+  # Image name of a stable release of Alertmanager, that works best with the current version of Prometheus
+  image: prom/alertmanager:v0.19.0 [ default = prom/alertmanager:v0.19.0 ]
+
+  # Path to the file containing the configMap for Alertmanager Deployment
+  config: path/to/config/file.yaml [ default = "" ]
+
+# This section contains configuration items specific to Prometheus component
+prometheus:
+
+  # Path to file containing Prometheus Rules
+  rules: path/to/rules/file.yaml [ default = "" ]
+
+  # Path to the swagger spec file that will be used to generate a service profile and provide per route metrics
+  swaggerSpecPath: path/to/swagger/spec [ default = "" ]
```

Comment:
Yaml is elaborate enough to store multiple line strings. So I guess we should be able to have those content directly specified here, rather an having links to file paths (circling back to the previous comment)